src/Panel/Model.php

```diff
@@ -95,14 +95,22 @@ public function dragTextType(string $type = null): string
      */
     public function icon(array $params = null): array
     {
-        $defaults = [
+        return array_merge($this->iconDefaults(), $params ?? []);
+    }
+
+    /**
+     * Default settings for icons
+     *
+     * @return array
+     */
+    public function iconDefaults(): array
```

Comment:
The return value is pretty specific to pages. I feel like we should make this method abstract and define it in `Panel\Page` instead. The disadvantage would be that we would need to define it in `Panel\Site` and `Panel\File` as well even though they use a completely different implementation for the icon (none or more complex respectively).